sound/soc/sof/ipc.c

```diff
@@ -267,6 +267,12 @@ static int tx_wait_done(struct snd_sof_ipc *ipc, struct snd_sof_ipc_msg *msg,
 				memcpy(reply_data, msg->reply_data,
 				       msg->reply_size);
 		}
+
+		/* re-enable dumps after successful IPC tx */
+		if (sdev->ipc_dump_printed) {
+			sdev->dbg_dump_printed = false;
+			sdev->ipc_dump_printed = false;
+		}
```

Comment:
I wasn't able to follow the commit message and comment. Usually an iPC timeout is rather fatal, audio will be broken until the next suspend-resume, it's not just a harmless 'hiccup'.

Reply:
@plbossart, this was my understanding as well. I had other patches (introducing the FW_CRASHED state) which would declare the firmware crashed if IPC timeout happens, which I thought was a correct way to handle this.
Then I saw sever logs where we had an IPC timeout reported (should be fatal) but consequent IPC messages were transmitted without issue and everything was just fine. It was a hiccup on the firmware side (scheduling spike).
My initial idea was to _block_ all further IPC tx after a timeout, but after hunting for more logs I have seen others where the timeout was followed by a recovery.

The reason for the re-enabling the dumps is exactly for this case, we got a timeout, we dump, then we will not show any dump for the consequent timeouts (unless all dump is enabled). If any consequent IPC tx manages to succeed then we re-enable the dumps to catch the next timeout.

No functionality change, just to give us more information, possibly hinting to a rootcause.

Reply:
I guess it depends where the IPC timeout happens. if it's for some random control that's probably ok. If it happens while restoring the state in the resume callback then it's as good as dead.

Reply:
> I guess it depends where the IPC timeout happens. if it's for some random control that's probably ok. If it happens while restoring the state in the resume callback then it's as good as dead.

Yes, that's correct. But it could be that in case of random control we are also dead. Based on what I saw in logs, it is not consistent.
I would expect the I'm dead interrupt from firmware in case it things go that bad, but for the case when we got timeout, but the firmware is still OK we want to capture the next time it happens again.
Let me try to clarify the commit message.